### ISSUE TITLE
[Bug]: Visual Studio hot reload broken

### ISSUE THREAD

**Issue by christosk92** [reactiveui/ReactiveUI#3675]
### Describe the bug 🐞

I’d like to know if this is indeed a bug or if it’s something on my end:

this issue occurs in both latest stable VS 17.8 and the latest preview (17.9)

I’m writing an app using WinUI 3 in WindowsAppSdk. 
hot reload works fine but the moment I add the ReactiveUI package to my project, it stops working. It doesn’t even give an error, it just straight up doesn’t compile the changes and I have to rebuild the app.

ive tried falling back to previous .NET and reactiveUI version but the problem persists, making me believe it’s an issue with visual studio or some compatibility error.

anyways I’d like to know if more people experienced this or if it’s just an issue on my end even thought I’ve spent hours debugging this issue now.

thanks!

### Step to reproduce

1. create a WinUI 3 app
2. Add ReactiveUI.WinUI to project 
3. Watch hot reload not working

### Expected behavior

Hot reload should work like it has

### IDE

Visual Studio 2022

### Operating system

Windows

### Version

17.8

**Comment by glennawatson** [comment]
We are working on a new release soon so see if that resolves your issues

**Comment by christosk92** [comment]
> We are working on a new release soon so see if that resolves your issues

Thanks for your reply.

In the meantime I will see if falling back to a previous version of VS resolves this issue.

**Comment by christosk92** [comment]
@glennawatson I was able to clone the Main branch of this repository and reference it in my app.
I can confirm this does indeed seem to fix the problem. Looking forward to an official release. Thanks !

**Comment by glennawatson** [comment]
Thanks. Be sometime this week. I got one optimisation I want to do before release.

**Comment by christosk92** [comment]
@glennawatson Eh... I think I spoke to soon. It seems that there were some incremental build shenanigans. After a full clean/rebuild the issue is back again. I have the source code running now so I might have a look at it to see what's wrong. I wasn't able to downgrade my Visual Studio (I have no idea how). Do you have any ideas perhaps why this issue might occur?

**Comment by christosk92** [comment]
@glennawatson Eh... I think I spoke to soon. It seems that there were some incremental build shenanigans. After a full clean/rebuild the issue is back again. I have the source code running now so I might have a look at it to see what's wrong. I wasn't able to downgrade my Visual Studio (I have no idea how). Do you have any ideas perhaps why this issue might occur?

**Comment by christosk92** [comment]
My apologies, i had some internet issues :(

**Comment by glennawatson** [comment]
Can you produce a minimum sample that causes the issue for you.

**Comment by christosk92** [comment]
Well it happens with any new WinAppSdk project.

The basic steps are; using the latest visual studio:
1) Create a C# WinUI with WindowsAppSdk app (not .NET Maui but actually native Windows)
2) Run the app, observe hot reload working properly. 
3) Add the ReactiveUI.WinUI package to the project.
4) clean and rebuild project.
5) Observe hot reload not working anymore.

A video:
https://streamable.com/u93oq9

**Comment by SoggyBottomBoy** [comment]
I just want to add I have similar issues, but with xaml. As soon as I add ReactiveUI as a nuget package then edit some xaml I get the following errors.

![image](https://github.com/reactiveui/ReactiveUI/assets/36346832/f491e91b-2134-4aa9-bde4-207d4c4ec59e)

The types of errors are:
`The specified value cannot be assigned to the collection. The following type was expected "UIElement"`
`The attached property "Column" can only be applied to types that derive from "UIElement"`
`The member "Spacing" is not recognized or is not accessible`

![image](https://github.com/reactiveui/ReactiveUI/assets/36346832/24f36b83-d329-4fbe-bb5e-ac2690423ae8)

The errors are generated when the xaml is edited at run time and appears to prevent xaml hot reload from working.

**Comment by christosk92** [comment]
Exactly the same issue for me too.

**Comment by parched** [comment]
Looks like this happened when I upgraded to 17.8

**Comment by ChrisPulman** [comment]
There is also similar problems with ASP.net, Console projects and VS2022 17.8, but no inclusion of ReactiveUI. I will try to debug the root cause and see if there's anything in ReactiveUI that is specifically causing the issue. From my projects the common factor is System.Reactive, either way hopefully I'll locate the issue.

**Comment by christosk92** [comment]
I actually managed to downgrade to Visual Studio 17.6 using https://learn.microsoft.com/en-us/visualstudio/releases/2022/release-history#release-dates-and-build-numbers

I can confirm that hot reload does indeed work with ReactiveUI. So it does seem like it's a regression from Visual Studio. Should we just close this issue here and report it to the vs community?

**Comment by christosk92** [comment]
FYI: I made a post about it on the VS developer community

https://developercommunity.visualstudio.com/t/SystemReactive-breaks-hot-reload-in-VS/10522361?sort=newest&q=hot+reload

**Comment by ChrisPulman** [comment]
Lets leave it open until VS2022 17.8.xxx is fixed, just for visibility for others.
This should avoid others raising the same issue. 
Thank you.

**Comment by ahardin** [comment]
FYI - this same issue appears to happen when using CommunityToolkit as well. I'm not using ReactiveUI, but I get the same lack of hot reload and the "errors" that @SoggyBottomBoy screenshotted.

You have to laugh to keep from crying. Windows App development is so miserably terrible that all these frameworks exist to make it less miserable. But you use the frameworks and VS breaks. Can't win.

**Comment by glennawatson** [comment]
Yeah little bit hard for us this one if it's a legitimate failure by visual studio. We can probably reach out to the teams if we get a decent repo. No guarantees tho

**Comment by christosk92** [comment]
Microsoft is getting very close to anti trust territory like the early 2000's.

**Comment by SoggyBottomBoy** [comment]
@christosk92 @glennawatson this bug appears fixed, I've just updated to vs 17.9 preview 5 and no longer experiencing this bug

**Comment by ChrisPulman** [comment]
> @christosk92 @glennawatson this bug appears fixed, I've just updated to vs 17.9 preview 5 and no longer experiencing this bug

Thank you for your feedback, glad the VS team finally got round to fixing it.
